agrirouter-sdk-java-tests/src/test/java/com/dke/data/agrirouter/test/messaging/rest/SendChunkedMessageTest.java

```diff
@@ -0,0 +1,115 @@
+import com.dke.data.agrirouter.api.dto.onboard.OnboardingResponse;
+import com.dke.data.agrirouter.api.enums.TechnicalMessageType;
+import com.dke.data.agrirouter.api.service.messaging.encoding.DecodeMessageService;
+import com.dke.data.agrirouter.api.service.messaging.encoding.EncodeMessageService;
+import com.dke.data.agrirouter.api.service.messaging.http.FetchMessageService;
+import com.dke.data.agrirouter.api.service.parameters.MessageHeaderParameters;
+import com.dke.data.agrirouter.api.service.parameters.MessageParameterTuple;
+import com.dke.data.agrirouter.api.service.parameters.PayloadParameters;
+import com.dke.data.agrirouter.api.service.parameters.SendMessageParameters;
+import com.dke.data.agrirouter.impl.common.MessageIdService;
+import com.dke.data.agrirouter.impl.messaging.SequenceNumberService;
+import com.dke.data.agrirouter.impl.messaging.encoding.DecodeMessageServiceImpl;
+import com.dke.data.agrirouter.impl.messaging.encoding.EncodeMessageServiceImpl;
+import com.dke.data.agrirouter.impl.messaging.rest.FetchMessageServiceImpl;
+import com.dke.data.agrirouter.impl.messaging.rest.SendMessageServiceImpl;
+import com.dke.data.agrirouter.test.AbstractIntegrationTest;
+import com.dke.data.agrirouter.test.Assertions;
+import com.dke.data.agrirouter.test.OnboardingResponseRepository;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Test;
+
+/** Test case to show the behavior for chunked message sending. */
+class SendChunkedMessageTest extends AbstractIntegrationTest {
+
+  public static final int EXPECTED_NUMBER_OF_CHUNKS = 3;
+
+  @Test
+  void
+      givenLargeContentMessageWhenSendingTheMessageToTheAgrirouterTheSdkShouldHelpToSendTheFileInMultipleChunks()
+          throws IOException, InterruptedException {
+
+    final EncodeMessageService encodeMessageService = new EncodeMessageServiceImpl();
+    final SendMessageServiceImpl sendMessageService = new SendMessageServiceImpl();
+    final OnboardingResponse onboardingResponse =
+        OnboardingResponseRepository.read(OnboardingResponseRepository.Identifier.FARMING_SOFTWARE);
+
+    MessageHeaderParameters messageHeaderParameters = new MessageHeaderParameters();
+    messageHeaderParameters.setTechnicalMessageType(TechnicalMessageType.ISO_11783_TASKDATA_ZIP);
+    messageHeaderParameters.setApplicationMessageId(MessageIdService.generateMessageId());
+    messageHeaderParameters.setApplicationMessageSeqNo(
+        SequenceNumberService.generateSequenceNumberForEndpoint(onboardingResponse));
+    messageHeaderParameters.setMode(Request.RequestEnvelope.Mode.PUBLISH);
+
+    PayloadParameters payloadParameters = new PayloadParameters();
+    payloadParameters.setValue(fakeLargeMessageContent());
```

Comment:
fakeLargeMessageContent does not provide a Base64-Encoded Content. Keep that in mind when fixing the issue in EncodeMessageServiceImpl.